streams/src/test/java/org/apache/kafka/streams/integration/KTableKTableForeignKeyJoinIntegrationTest.java

```diff
@@ -244,21 +248,21 @@ public void doJoinFromLeftThenDeleteLeftEntity() {
                     );
                 }
             }
-            // Now delete one LHS entity such that one delete is propagated down to the output.
 
-            left.pipeInput("lhs1", (String) null, baseTimestamp + 6);
+            // Now delete one LHS entity such that one delete is propagated down to the output.
+            left.pipeInput("lhs1", null, baseTimestamp + 6);
             assertThat(
-                outputTopic.readKeyValuesToMap(),
-                is(mkMap(
-                    mkEntry("lhs1", null)
+                outputTopic.readKeyValuesToList(),
+                is(Collections.singletonList(
+                    KeyValue.pair("lhs1", null)
                 ))
             );
             if (rejoin) {
                 assertThat(
-                    rejoinOutputTopic.readKeyValuesToMap(),
-                    is(mkMap(
-                        mkEntry("lhs1", null)
-                    ))
+                    rejoinOutputTopic.readKeyValuesToList(),
+                    hasItem(
```

Comment:
Not testing all items because the test is not deterministic, it may or may not contain `("lhs1", "rejoin((lhsValue1|rhs1,rhsValue1),null)")` depending on whether it's the left hand or the right hand side of the final join which propagates first. 

`rejoin`  is computed this way in the test:

                   ___________LEFT
                  |            |
                  |    RIGHT   | 
                  |     |      |
                  \     /      /  
                   \   /      /
                    \ /      /
                  FKJoin    /
                     \     /
                      \   /
                      Rejoin